pkg/mimir/modules.go

```diff
@@ -828,6 +828,15 @@ func (t *Mimir) initCompactor() (serv services.Service, err error) {
 func (t *Mimir) initStoreGateway() (serv services.Service, err error) {
 	t.Cfg.StoreGateway.ShardingRing.ListenPort = t.Cfg.Server.GRPCListenPort
 
+	// TODO: Remove in Mimir 2.12.
+	// Take value from Indexheader DeprecatedConfig if it is set
+	if t.Cfg.BlocksStorage.BucketStore.DeprecatedIndexHeaderLazyLoadingEnabled {
```

Comment:
Lazy loading is enabled by default. I think if someone is trying to disable lazy loading they can only do it via the old config flag. Can we use whichever flag differs from the default?

Reply:
Updated in 22a5d0ccfd11d9712385a06822ca52abb4bc19e7